packages/legacy/core/App/hooks/proof-request-templates.ts

```diff
@@ -1,18 +1,130 @@
-  const { proofRequestTemplates } = useConfiguration()
-  const [store] = useStore()
-  return (
-    proofRequestTemplates &&
-    proofRequestTemplates(store.preferences.acceptDevCredentials).find((template) => template.id === templateId)
+const calculatePreviousYear = (yearOffset: number) => {
+  const pastDate = new Date()
+  pastDate.setFullYear(pastDate.getFullYear() + yearOffset)
+  return parseInt(pastDate.toISOString().split('T')[0].replace(/-/g, ''))
+}
+
+export const applyTemplateMarkers = (templates: any): any => {
+  const markerActions: { [key: string]: (param: string) => string } = {
+    now: () => Math.floor(new Date().getTime() / 1000).toString(),
+    currentDate: (offset: string) => calculatePreviousYear(parseInt(offset)).toString(),
+  }
+  let templateString = JSON.stringify(templates)
+  const markers = [...templateString.matchAll(/"@\{(\w+)(?:\((\S*)\))?\}"/gm)]
+  const uniqueMarkers = [...new Set(markers.map((marker) => JSON.stringify(marker)))].map((marker) =>
+    JSON.parse(marker)
   )
+  uniqueMarkers.forEach((marker) => {
+    const markerValue = markerActions[marker[1] as string](marker[2])
+    templateString = templateString.replaceAll(marker[0], markerValue)
+  })
+  return JSON.parse(templateString)
+}
+
+export const applyDevRestrictions = (templates: ProofRequestTemplate[]): ProofRequestTemplate[] => {
+  return templates.map((temp) => {
+    return {
+      ...temp,
+      payload: {
+        ...temp.payload,
+        data: (temp.payload as AnonCredsProofRequestTemplatePayload).data.map((data) => {
+          return {
+            ...data,
+            requestedAttributes: data.requestedAttributes?.map((attr) => {
+              return {
+                ...attr,
+                restrictions: [...(attr.restrictions ?? []), ...(attr.devRestrictions ?? [])],
+                devRestrictions: [],
+              }
+            }),
+            requestedPredicates: data.requestedPredicates?.map((pred) => {
+              return {
+                ...pred,
+                restrictions: [...(pred.restrictions ?? []), ...(pred.devRestrictions ?? [])],
+                devRestrictions: [],
+              }
+            }),
+          }
+        }),
+      },
+    }
+  })
+}
+
+export class RemoteProofBundleResolver implements ProofBundleResolverType {
+  private remoteServer
+  private templateData: ProofRequestTemplate[] | undefined
+
+  public constructor(indexFileBaseUrl: string) {
+    this.remoteServer = axios.create({
+      baseURL: indexFileBaseUrl,
+    })
+  }
+  public async resolve(acceptDevRestrictions: boolean): Promise<ProofRequestTemplate[] | undefined> {
+    if (this.templateData) {
+      let templateData = this.templateData
+      if (acceptDevRestrictions) {
+        templateData = applyDevRestrictions(templateData)
+      }
+      return Promise.resolve(applyTemplateMarkers(templateData))
+    }
+    return this.remoteServer.get('proof-templates.json').then((response) => {
```

Comment:
how is this file being cached?

Reply:
right now it's just saving the templates in temporary memory, I'm going to implement saving the templates to local storage

Reply:
so right now it will fetch the templates once when is starts up, but I'll implement some caching